addons/website/models/theme_models.py

```diff
@@ -275,6 +275,9 @@ class IrUiView(models.Model):
     theme_template_id = fields.Many2one('theme.ir.ui.view', copy=False)
 
     def write(self, vals):
+        # We only do this if we are in a module update state
+        if not self.pool._init:
```

Comment:
This is a bit tricky as this is related to the tests, which tries to simulate the best they can the flow they are testing.

In this case it is because https://github.com/odoo/odoo/commit/1eb7c577ecb31c2f0876b6dd76769e2b47802cb0#diff-86206ee83b8963a647f6373486c147e25cf21204a3fc812368d3dad587c5dd99R1316 is simulating a module update, which in real case is going through this `write()` as `pool._init` is True.
But in the test, it is not True, as this is not a real update (just simulating).

The workaround would be to do
```
import threading
# [...]
test_mode = getattr(threading.currentThread(), 'testing', False)
if not (test_mode or self.pool._init):
```
That's not exactly correct as the tests will now avoid this condition, acting as it was doing an update, but that's not that bad IMHO, as it was like that without your PR already.

The other solution might be to do in the test line I mentioned `with_context(simulate_module_update=True)` and here check if we have that key context. As you prefer.

I am not a big fan of writing code related to tests in our regular codebase, but in this case I don't see a better way, if we want to keep the test.

Let me know if you prefer that I finish this PR, both ways are ok to me

Reply:
@xmo-odoo May I ask your opinion here about what I said?
Is there a better way you can think of?

Reply:
> May I ask your opinion here about what I said? Is there a better way you can think of?

Not really, I can't think of something significantly better.

The code comment is pretty unhelpful though, it says what the code already says which is redundant, *why* this code is the way it is would be more useful aka the purpose of the override, and of bypassing it.

I think the issue caused by the override would be useful too, "there's an error" is not really helpful to the reader of the commit message, what the error is and why it happens would be more useful: it looks like the override tries to be careful when performing comparisons, so one could expect regular editions to work fine (I assume that is what happened originall).

Also while at it, the return values of the sub-writes in the overide are useles (and uselessly combined): `write` should just return a static `True`. In case of errors, it just raises an exception.

Reply:
Yes, i will change it. The new message is quite easy to understand. Sorry for my limited English, so the commit message is not detailed.